pkg/utils/exec/exec.py

```diff
@@ -0,0 +1,52 @@
+
+from kubernetes.stream import stream
+
+# PodDetails contains all the required variables to exec inside a container
+class PodDetails(object):
+	def __init__(self, PodName=None, Namespace=None, ContainerName=None):
+		self.PodName      = PodName
+		self.Namespace       = Namespace
+		self.ContainerName = ContainerName
+
+# checkPodStatus verify the status of given pod & container
+def checkPodStatus(pod , containerName):
+
+    if pod.status.phase.lower() != "running":
+        return ValueError("%s pod is not in running state, phase: %s", pod.Name, pod.Status.Phase)
+
+    for container in pod.status.container_statuses:
+        if container.name == containerName and container.state.terminated != None:
```

Comment:
why are we checking container state for termination? can't we use `!container.Ready`?